helm-tests/validate_typha_test.go

```diff
@@ -0,0 +1,106 @@
+package main_test
+
+import (
+	"testing"
+)
+
+// This file holds all of the functions for validating
+// expected values on the Typha deployment.
+
+func TestTypha(t *testing.T) {
+	t.Run("typha enabled with kubernetes datastore", func(t *testing.T) {
+		testYml := `datastore: kdd
+typha:
+  enabled: true`
+
+		resources, err := render(testYml)
+		if err != nil {
+			t.Error(err, "Error while rendering resources with typha enabled and datastore set to a kubernetes datastore")
+		}
+
+		// Validate that the typha resources exist
+		if resources.calicoTypha == nil {
+			t.Error(err, "calico-typha deployment should exist when typha is enabled with a Kubernetes datastore")
+		}
+
+		if resources.calicoTyphaService == nil {
+			t.Error(err, "calico-typha service should exist when typha is enabled with a Kubernetes datastore")
+		}
+
+		if resources.calicoTyphaPDB == nil {
+			t.Error(err, "calico-typha pod disruption budget should exist when typha is enabled with a Kubernetes datastore")
+		}
+	})
+
+	t.Run("typha disabled with kubernetes datastore", func(t *testing.T) {
+		testYml := `datastore: kdd
```

Comment:
nit: name this `valuesYml` to be more clear.

Reply:
Personally, I find this style of inline yaml definition to be really hard to read and also really hard to write properly.

Can we do something like define a go struct, so values can be defined in golang and then rendered as a yaml?

e.g.:

```go
values := Values{datastore: "kdd"}
valuesBytes, err := yaml.Marshal(values)
```